replicatedhq/troubleshoot
# feat: Add regex host analyser

## Description, Motivation and Context

This anaylser is the same as the [in-cluster regex anaylser](https://troubleshoot.sh/docs/analyze/regex/). You pass in search expressions to find values in files collected in a bundle 

Manual test cases
* Passed analyser
```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: example
spec:
  hostCollectors:
    - run:
        collectorName: "localhost-ips"
        command: "sh"
        args: ["-c", "host localhost"]
  hostAnalyzers:
    - textAnalyze:
        checkName: Check if localhost resolves to 127.0.0.1
        fileName: host-collectors/run-host/localhost-ips.txt
        regex: 'localhost has address 127.0.0.1'
        outcomes:
          - fail:
              when: "false"
              message: "'localhost' does not resolve to 127.0.0.1 ip address. Please check your local configuration settings"
          - pass:
              when: "true"
              message: "'localhost' resolves to 127.0.0.1 ip address"
```

<img width="1046" alt="image" src="https://github.com/replicatedhq/troubleshoot/assets/681087/256f780c-8be0-4cb5-8990-d72c24443abe">

* Failed analyser - In this test case we inject the results using [static data](https://troubleshoot.sh/docs/collect/data/) cause I could not get a test environment to drop `127.0.0.1` from the local DNS cache.
```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: example
spec:
  collectors:
    - data:
        name: host-collectors/run-host/localhost-ips.txt
        data: |
          localhost has IPv6 address ::1
  hostAnalyzers:
    - textAnalyze:
        checkName: Check if localhost resolves to 127.0.0.1
        fileName: host-collectors/run-host/localhost-ips.txt
        regex: 'localhost has address 127.0.0.1'
        outcomes:
          - fail:
              when: "false"
              message: "'localhost' does not resolve to 127.0.0.1 ip address"
          - pass:
              when: "true"
              message: "'localhost' resolves to 127.0.0.1 ip address"
```

<img width="1046" alt="image" src="https://github.com/replicatedhq/troubleshoot/assets/681087/2442b992-6e58-4390-aaae-e85d2096a9cd">




Fixes: #1263 

Docs PR: https://github.com/replicatedhq/troubleshoot.sh/pull/513

## Checklist

- [x] New and existing tests pass locally with introduced changes.
- [x] The commit message(s) are informative and highlight any breaking changes
- [x] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
